### ISSUE TITLE
Calibration resulting in significant slope shift

### ISSUE THREAD

**Issue by old-square-eyes** [NightscoutFoundation/xDrip#318]
Currently, a calibration can result in a significant slope shift. Sometimes this might be desired (i.e. bringing in a slope to where it should be, perhaps after a bump on a sensor).

However, twice recently I have calibrated, and it has had an undesirable effect on my slope. I haven't noticed and it has resulted in xDrip interpreting that I an 1-2 mmol/L higher than I actually was.

Both times I woke up to test and that calibration (I use auto calibrate) pulled my chart to the left significantly.

I wonder if there could be a toast / alert to warn that this has happened? So the user can review the chart and potentially reconsider the accuracy of the calibration and otherwise act on it.

![screenshot_20180226-084559](https://user-images.githubusercontent.com/25760491/36648824-0fc581ce-1a90-11e8-96e7-a19811362986.png)
![screenshot_20180226-084619](https://user-images.githubusercontent.com/25760491/36648825-0ff5ed64-1a90-11e8-8d29-000173125640.png)

**Comment by Navid200** [comment]
@old-square-eyes 
When you calibrate, either the slope will be affected or the intercept or both.

I understand you like to have a warning when the slope is changed too much.  
But, someone else may also like to get a warning when the intercept has changed too much.
I think the solution is to just check both intercept and slope after every calibration to be safe.  
If you make that a habit, you will not need a warning.  Do you agree?

**Comment by old-square-eyes** [comment]
Yes, I should have said slope and calibration. Though many people don't understand it so simple language is probably best. "Calibration has changed significantly, are you sure you wish to proceed", or something like that.

**Comment by Navid200** [comment]
Considering you have been a part of the community longer than me, I am just making suggestions for your consideration.
In my opinion, we diabetics need alarms to tell us that our glucose is high or low.

But, I think we can survive if we don't get an alarm when the calibration slope or intercept has changed.  That happens when we calibrate and we should always check the result of the calibration.

So, my question for you is this.  Are you sure you want to ask for a new alarm?  Considering that it has to be developed and there are so many other feature requests already open.  I hope you agree to allow this one to go.
Otherwise, would you be willing to let me label this as postponed?  So, if we get to a point when other features are addressed, we can reconsider and reopen it.

**Comment by old-square-eyes** [comment]
Couple of things. Agree that more alarms is potentially annoying for users, and extra dev work. Seconly the number of people using raw data and calibrating on xDrip is shrinking due to the new Dexcom transmitters (not sure what other sensors this feature would service). It's a very old ticket and I use Native myself now. Though, logs and UX in xDrip does not quite gel that well with Native users - impossible slope/initial calibration experience/errors etc, which are irellivant for Native users - but that's another Issue).

This ticket was from when I was using auto calibration feature in xDrip. By and large it does a good job of automatically accepting. But as described there were times when unwittingly (and other automatically) a big, and incorrect, cxalibration shift was entered/accepted. The audio alarm suggested was for times when auto calibrate accepted something that resulted in the shift... the toast was for somone manually entering and accepting a calibration, and just servs as a gentle reminder to check thier calibration graph and if the new plot is way off, to consider the reasons for that and possibly remove and retry the calibration.

**Comment by old-square-eyes** [comment]
from a safety perspective, I personally, only calibrate once a day, even on Raw. SO a big shift undetected can result in a day of wrong readings. High or low, it's not great.

**Comment by Navid200** [comment]
There has been no development on this.
I would like to either close the issue or convert it to a discussion.

Please tell me if you have a preference.
Thanks
